contracts/contracts/RequestManager.sol

```diff
@@ -510,17 +510,42 @@ contract RequestManager is Ownable, LpWhitelist, RestrictedCalls, Pausable {
     /// of native tokens staked by the dishonest claimer.
     ///
     /// @param claimId The claim ID.
-    /// @return The address of the deposit receiver.
+    /// @return The claim stakes receiver.
     function withdraw(uint96 claimId)
         external
         validClaimId(claimId)
         returns (address)
+    {
+        return withdraw(msg.sender, claimId);
+    }
+
+    /// Withdraw the deposit that the request submitter left with the contract,
+    /// as well as the staked native tokens associated with the claim.
+    ///
+    /// This function is called on behalf of claimResolver. Only the claimResolver address
+    /// may receive the funds if he is the winner of the challenge or the claim is valid.
+    ///
+    /// In case the caller of this function is a challenger that won the game,
+    /// they will only get their staked native tokens plus the reward in the form
+    /// of full (sole challenger) or partial (multiple challengers) amount
+    /// of native tokens staked by the dishonest claimer.
+    ///
+    /// @param claimId The claim ID.
+    /// @param claimResolver The claim resolver.
+    /// @return The claim stakes receiver.
+    function withdraw(address claimResolver, uint96 claimId)
```

Comment:
I think the name `claimResolver` is quite confusing, especially when reading the code of `resolveClaim`. I would suggest using a different name that more directly indicates that this is the address that will get the stakes in case claim is invalid. E.g. `stakeRecipient` or something like that.

Reply:
I agree.

`stakeRecipient` is also not a good name, because resolveClaim will output the claimReceiver which will receive the stakes. In the case that the claimer actually wins, `stakeRecipient` can be set to any address but the claimer will receive the stakes. 
Do you have any other suggestion?

Reply:
> `stakeRecipient` is also not a good name, because resolveClaim will output the claimReceiver which will receive the stakes. In the case that the claimer actually wins, `stakeRecipient` can be set to any address but the claimer will receive the stakes. Do you have any other suggestion?

Sadly, I can't think of anything better, at least for now.

Note that the confusion you describe arises only if you're acting dishonestly and calling the function.
In case you're honest, everything is still fine because
- if you as a claimer call this, you name yourself the state recipient because you're expecting to get your stake back
- if you as a challenger call this, similar to above, you name yourself the state recipient because you're expecting to get your stake back
- if you're anyone else (regardless of participation in the game), you name the stake recipient, and they will get their stake _if they are entitled to_